bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/JsonUtils.java

```diff
@@ -307,7 +306,7 @@ private static Object convertToJsonType(Object value, List<TypeValuePair> unreso
             return null;
         }
         Type sourceType = TypeChecker.getType(value);
-        if (TypeUtils.getReferredType(sourceType).getTag() <= TypeTags.BOOLEAN_TAG && TypeChecker.checkIsType(value,
+        if (TypeUtils.getConclusiveType(sourceType).getTag() <= TypeTags.BOOLEAN_TAG && TypeChecker.checkIsType(value,
```

Comment:
What about a name like `getRepresentedType`? Or implied type. Does conclusive really mean what we want to convey?

Reply:
+1 for `getRepresentedType`